vendor/github.com/aead/poly1305/poly1305_ref.go

```diff
@@ -34,7 +32,8 @@ func Sum(msg []byte, key [32]byte) [TagSize]byte {
 		msg = msg[n:]
 	}
 	if len(msg) > 0 {
-		off = copy(block[:], msg)
+		var block [TagSize]byte
+		off := copy(block[:], msg)
```

Comment:
if len(msg) ever is TagSize, then the following statement `block[off] = 1` would cause a buffer overrun -- may want to protect against this?